### PR TITLE
Refactor FXIOS-5383 [v108] ForegroundBVC made optional and sentry logs

### PR DESCRIPTION
**Pull request by nishant2718**
# [FXIOS-5383](https://mozilla-hub.atlassian.net/browse/FXIOS-5383) | #12601 

TLDR: ForegroundBVC made optional & sentry logging added. 

## Details
With the transition to [a single scened app](https://github.com/mozilla-mobile/firefox-ios/pull/12241), we've, surprisingly, started seeing crashes caused by times where a scene doesn't exist, [even five seconds](https://github.com/mozilla-mobile/firefox-ios/blob/63fead9be61920f5a724f2e8ac9d3d4f29e73ecb/Client/Application/AppDelegate.swift#L108) into the app session. 

Until we can investigate how and why this happens, we're going to temporarily make foregroundBVC return an optional. We are aware that this will break functionality where BVC is expected to be non-nil. However, these cases should be very rare, and we'll have some logs for it by then.

### CODE REVIEW

File: Client/Frontend/Settings/AppSettingsOptions.swift [hunk 1]
```diff
@@ -820,7 +823,9 @@ class ShowIntroductionSetting: Setting {
 
     override func onClick(_ navigationController: UINavigationController?) {
         navigationController?.dismiss(animated: true, completion: {
-            BrowserViewController.foregroundBVC().presentIntroViewController(true)
+            // TODO: Temporary.
+            // This instance of foregroundBVC is going to be revisited after having enough telemetry of ShowTour.
```
Comment (lmarceau) [inline Client/Frontend/Settings/AppSettingsOptions.swift:827]: Just to be aware, this telemetry is already out in prod or will be soon?

**Review by lmarceau — APPROVED** [review]

**Comment by nishant2718** [comment]
Test failure on a possibly flaky test. It passes locally, and I ran HistoryTests a couple times to make sure.

**Comment by nishant2718** [comment]
SwiftLint fixes - so will skip BR to save creds.
